### ISSUE TITLE
export-ing a string with quotes causes SC2089+SC2090 warning

### ISSUE THREAD

**Issue by joanbm** [koalaman/shellcheck#2064]
#### For bugs
- Rule Id: SC2089 and SC2090
- Shellcheck version: 0.7.1 (Arch Linux)
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/bash
TVAR="Smiths' Best Bread"
export TVAR
env | grep TVAR # Works
```

#### Here's what shellcheck currently says:

```
In test.sh line 2:
TVAR="Smiths' Best Bread"
      ^----------------^ SC2089: Quotes/backslashes will be treated literally. Use an array.


In test.sh line 3:
export TVAR
       ^--^ SC2090: Quotes/backslashes in this variable will not be respected.

For more information:
  https://www.shellcheck.net/wiki/SC2089 -- Quotes/backslashes will be treate...
  https://www.shellcheck.net/wiki/SC2090 -- Quotes/backslashes in this variab...
```

#### Here's what I wanted or expected to see:

Nothing (no warnings).

#### Notes

There may be something behind this warning, but the documentation for the SC2089/SC2090 warnings does not give me any hint why of what's wrong. I expect the variable to be exported as-is, including the quotes (e.g. so I can pass it to a different program), and this seems to be what the script is doing when running it. I tested various further examples with single quotes, double quotes and backslashes and there seems to be no ill effect from exporting variables containing those.

Using an array here as recommended by SC2089 does not seem useful. I couldn't find anything online, on `man export` and `man bash` that hints that `export`-ing a variable with quotes may be the wrong thing to do.

If I replace `export TVAR` with `export TVAR="$TVAR"` instead, the warning disappears.

**Comment by GfEW** [comment]
> FYI, for me to make the warning disappear, it is enough to change
> 
> export TVAR
> 
> with
> 
> export "TVAR"

Just for the record:

**variant (a):** `export TVAR`
- doesn't trigger SC2034 **# why not?**
- triggers SC2089 **# why?**
- triggers SC2090 **# why?**

**variant (b):** `export "TVAR"`
- triggers SC2034
- doesn't trigger SC2089
- doesn't trigger SC2090

**variant (c):** `export TVAR="$TVAR"`
- doesn't trigger SC2034 **# why not?**
- doesn't trigger SC2089
- doesn't trigger SC2090

AFAICT, these differences, in particular the **marked** outcomes, make no actual sense - neither code/style wise nor in terms of being helpful at teaching important concepts.

The user just shouldn't be required to any of
- working around SC2034, SC2089, SC2090 by choosing one variant over another
- placing
`# shellcheck disable=SC2089,SC2090`
before every string containing a quote _in order to keep those warnings active for cases where they **do** serve a purpose_
- having the same pointless warnings wear their alertness over and over again
- disabling them globally

but rather be warned specifically where it matters.
